internal/resources/metal/organization/datasource.go

```diff
@@ -0,0 +1,125 @@
+package organization
+
+import (
+	"context"
+	"fmt"
+
+	equinix_errors "github.com/equinix/terraform-provider-equinix/internal/errors"
+	"github.com/equinix/terraform-provider-equinix/internal/framework"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/packethost/packngo"
+)
+
+func NewDataSource() datasource.DataSource {
+	return &DataSource{
+		BaseDataSource: framework.NewBaseDataSource(
+			framework.BaseDataSourceConfig{
+				Name: "equinix_metal_organization",
+			},
+		),
+	}
+}
+
+type DataSource struct {
+	framework.BaseDataSource
+}
+
+func (r *DataSource) Schema(
+	ctx context.Context,
+	req datasource.SchemaRequest,
+	resp *datasource.SchemaResponse,
+) {
+	resp.Schema = dataSourceSchema(ctx)
+}
+
+func (r *DataSource) Read(
+	ctx context.Context,
+	req datasource.ReadRequest,
+	resp *datasource.ReadResponse,
+) {
+	r.Meta.AddFwModuleToMetalUserAgent(ctx, req.ProviderMeta)
+	client := r.Meta.Metal
+
+	// Retrieve values from plan
+	var data DataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Extract the ID of the resource from the state
+	orgId := data.OrganizationID
+	name := data.Name
+
+	if !orgId.IsNull() || !name.IsNull() {
+		resp.Diagnostics.AddError(
+			"You must provide either orgId OR Name", "",
+		)
+	}
```

Comment:
I think we could replace these lines with an update to `datasource_schema.go` to specify this for the `organization_id` attribute:
```
Validators: []validator.String{
  stringvalidator.ExactlyOneOf(path.MatchRoot("organization_id"), path.MatchRoot("name")),
}
```